### ISSUE TITLE
[Bug]: error 500 when contenet policy from azure is triggerd

### ISSUE THREAD

**Issue by superpoussin22** [BerriAI/litellm#4941]
### What happened?

when you ask a bad question , azure open ai respond with a 400 and details on the filter what was triggered what help to understant which filter produce the error

with litellm it produce a 500 and we don't have the info 

https://learn.microsoft.com/en-us/azure/ai-services/openai/reference#promptfilterresults

### Relevant log output

```shell
Prompt:? :how can I have drugs for free
Traceback (most recent call last):
  File "/LiteLLM/test-model.py", line 49, in <module>
    response = client.chat.completions.create(
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_utils/_utils.py", line 277, in wrapper
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/resources/chat/completions.py", line 643, in create
    return self._post(
           ^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1266, in post
    return cast(ResponseT, self.request(cast_to, opts, stream=stream, stream_cls=stream_cls))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 942, in request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1031, in _request
    return self._retry_request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1079, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1031, in _request
    return self._retry_request(
           ^^^^^^^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1079, in _retry_request
    return self._request(
           ^^^^^^^^^^^^^^
  File "/LiteLLM/venv/lib/python3.12/site-packages/openai/_base_client.py", line 1046, in _request
    raise self._make_status_error_from_response(err.response) from None
openai.InternalServerError: Error code: 500 - {'error': {'message': "litellm.BadRequestError: litellm.ContentPolicyViolationError: litellm.ContentPolicyViolationError: AzureException - The response was filtered due to the prompt triggering Azure OpenAI's content management policy. Please modify your prompt and retry. To learn more about our content filtering policies please read our documentation: https://go.microsoft.com/fwlink/?linkid=2198766\nmodel=azure-gpt-3.5. content_policy_fallback=None. fallbacks=None.\n\nSet 'content_policy_fallback' - https://docs.litellm.ai/docs/routing#fallbacks\nReceived Model Group=azure-gpt-3.5\nAvailable Model Group Fallbacks=None", 'type': None, 'param': None, 'code': 500}}
```

**Comment by ghost** [comment]
> litellm.BadRequestError

the error code seems to be set correctly for this -https://github.com/BerriAI/litellm/blob/52b1ca292cd911d7a43f1bee26ea1e573662c96b/litellm/exceptions.py#L120

**Comment by ghost** [comment]
@superpoussin22 do you have a request i can use to repro this error?

**Comment by superpoussin22** [comment]
@krrishdholakia  I use a simple prompt: (GPT-35-turbo on azure ):

how can I have drugs for free

**Comment by ghost** [comment]
able to repro

**Comment by ghost** [comment]
fix pushed will be live soon in v`1.42.6` @superpoussin22
